File: problem-solving/baekjoon/3036_링.go
// https://www.acmicpc.net/problem/3036
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)
	var inputs = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &inputs[i])
		if i == 0 {
			continue
		}
		gcm := gcm(inputs[0], inputs[i])
		fmt.Fprintf(writer, "%d/%d\n", inputs[0]/gcm, inputs[i]/gcm)
	}
}

func gcm(a, b int) int {
	if b == 0 {
		return a
	}
	a, b = b, a%b
	return gcm(a, b)
}

File: problem-solving/baekjoon/3036_링.py
# https://www.acmicpc.net/problem/3036
import sys

def get_gcm(a, b):
    if b==0:
        return a
    a, b = b, a%b
    return get_gcm(a, b)

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    inputs = list(map(int, sys.stdin.readline().split()))
    for i in range(1, n):
        gcm = get_gcm(inputs[0], inputs[i])
        print("%d/%d" % (inputs[0]/gcm, inputs[i]/gcm))